e2e/test_operator/operator_test.go

```diff
@@ -2143,4 +2145,115 @@ var _ = Describe("Operator", Label("e2e", "pr"), func() {
 			Expect(fdbCluster.WaitForReconciliation()).NotTo(HaveOccurred())
 		})
 	})
+
+	When("a FDB pod is partitioned from the Kubernetes API", func() {
+		var selectedPod corev1.Pod
+		var exp *fixtures.ChaosMeshExperiment
+		var newCustomParameters, initialCustomParameters fdbv1beta2.FoundationDBCustomParameters
+		var initialRestarts int
+
+		BeforeEach(func() {
+			// If we are not using the unified image, we can skip this test.
+			if !fdbCluster.GetCluster().GetUseUnifiedImage() {
+				Skip("The sidecar image doesn't require connectivity to the Kubernetes API")
+			}
+
+			if !factory.ChaosTestsEnabled() {
+				Skip("Chaos tests are skipped for the operator")
+			}
+
+			selectedPod = fixtures.RandomPickOnePod(fdbCluster.GetStoragePods().Items)
+			for _, status := range selectedPod.Status.ContainerStatuses {
+				initialRestarts += int(status.RestartCount)
+			}
+
+			var kubernetesServiceHost string
+			Eventually(func(g Gomega) error {
+				std, _, err := factory.ExecuteCmdOnPod(
+					&selectedPod,
+					fdbv1beta2.MainContainerName,
+					"printenv KUBERNETES_SERVICE_HOST",
+					false,
+				)
+
+				g.Expect(std).NotTo(BeEmpty())
+				kubernetesServiceHost = strings.TrimSpace(std)
+
+				return err
+			}, 5*time.Minute).ShouldNot(HaveOccurred())
+
+			exp = factory.InjectPartitionWithExternalTargets(fixtures.PodSelector(&selectedPod), []string{kubernetesServiceHost})
+			// Make sure that the partition takes effect.
+			Eventually(func() error {
+				_, _, err := factory.ExecuteCmdOnPod(
+					&selectedPod,
+					fdbv1beta2.MainContainerName,
+					fmt.Sprintf("nc -vz -w 2 %s 443", kubernetesServiceHost),
+					false,
+				)
+
+				return err
+			}).WithTimeout(2 * time.Minute).WithPolling(1 * time.Second).Should(HaveOccurred())
+
+			// Rollout a new knob to check the behaviour in such a case.
+			initialCustomParameters = fdbCluster.GetCustomParameters(
+				fdbv1beta2.ProcessClassStorage,
+			)
+
+			newCustomParameters = append(
+				initialCustomParameters,
+				"knob_max_trace_lines=1000000",
+			)
+
+			Expect(
+				fdbCluster.SetCustomParameters(
+					fdbv1beta2.ProcessClassStorage,
+					newCustomParameters,
+					false,
+				),
+			).NotTo(HaveOccurred())
+		})
+
+		It("should keep the pod up and running", func() {
+			// Make sure the partitioned Pod is not able to update its annotation.
+			Eventually(func() *int64 {
+				selectedProcessGroupID := fixtures.GetProcessGroupID(selectedPod)
+				for _, processGroup := range fdbCluster.GetCluster().Status.ProcessGroups {
+					if processGroup.ProcessGroupID != selectedProcessGroupID {
+						continue
+					}
+
+					return processGroup.GetConditionTime(fdbv1beta2.IncorrectConfigMap)
+				}
+
+				return nil
+			}).WithTimeout(5 * time.Minute).WithPolling(1 * time.Second).MustPassRepeatedly(10).ShouldNot(BeNil())
```

Comment:
should we be checking that the timestamp isn't changing?

Reply:
We could be doing that, but it should be enough to check that the condition is present.